content/blog/testthat-3-0-0/index.Rmd

```diff
@@ -0,0 +1,113 @@
+
+-   `context()` is now deprecated.
+    It hasn't been recommended for some time since it usually duplicates data also present in the file name.
+
+-   `expect_identical()` and `expect_equal()` use `waldo::compare()` to compare actual and expected results.
+    This should give considerably more informative output for test failures, but `waldo::compare()` is not 100% compatible with `all.equal()` (which previously powered `expect_equal()`).
+    There are a few differences due a bug in testthat's comparison of floating point values and subtle differences when comparing environments stored in attributes (common in modelling functions); see the vignette for full details.
+
+    `expect_equivalent()` is now deprecated because it's equivalent to `expect_equal(ignore_attr = TRUE)` (which is generally not recommended, anyway, because it's such a blunt tool).
+
+-   `expect_error()`, `expect_warning()`, `expect_message()`, and `expect_condition()` now all use the same underlying logic: they capture the first condition that matches `class`/`regexp` and allow anything else to bubble up.
+    This behaviour pairs particularly well with upcoming changes to the pipe in [magrittr 2.0.0](/blog/2020/08/magrittr-2-0/).
+
+-   Messages are no longer automatically silenced.
+    You'll now need to use `suppressMessages()` to hide unimportant messages or `expect_messsage()` to catch important messages.
+
+-   `test_that()` now sets a number of options and env vars to make output as reproducible as possible.
+    Many of these options were previously set in various places (including `devtools::test()`, `test_dir()`, `test_file()`, and `verify_output()`) but they now been centralised in to `local_test_context()`.
+
+## Snapshot testing
+
+The goal of a unit test is to record the expected output of a function using code.
+This is a powerful technique because because not only does it ensure that code doesn't change unexpectedly, it also expresses the desired behaviour in a way that a human can understand.
+However, it's not always convenient to record the expected behaviour with code.
+Some challenges include:
+
+-   Text output that includes many characters like quotes and newlines that require special handling in a string.
+
+-   Output that is large, making it painful to define the reference output, and bloating the size of the test file and making it hard to navigate.
+
+-   Cases where you want to record a mix of printed output, messages, warnings, or errors.
+
+-   Binary formats like plots or images, which are very difficult to describe in code: i.e. the plot looks right, the error message is useful to a human, the print method uses colour effectively.
+
+For these situations, testthat provides an alternative mechanism: snapshot tests.
+Instead of using code to describe expected output, snapshot tests (also known as [golden tests](https://ro-che.info/articles/2017-12-04-golden-tests)) record results in a separate human readable file.
```

Comment:
```suggestion
Instead of using code to describe expected output, snapshot tests (also known as [golden tests](https://ro-che.info/articles/2017-12-04-golden-tests)) record results in a separate human-readable file.
```